### ISSUE TITLE
tf.keras.layers.BatchNormalization computes moving variances inconsistently when `fused=True` vs. when `fused=False` 

### ISSUE THREAD

**Issue by canbakiskan** [tensorflow/tensorflow#55545]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.8.0 (gpu)
- Python version: 3.10.2
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 11.5/8.3.1
- GPU model and memory: 1080 Ti 12GB

**Describe the current behavior**

`tf.keras.layers.BatchNormalization` computes moving variances inconsistently when `fused=True` vs when `fused=False`. When `fused=False` it computes it with 

             next_variance=old_variance*momentum+batch_variance*(1-momentum)

whereas when `fused=True` it computes it with: 

             next_variance=old_variance*momentum+batch_variance*(1-momentum)*bessel_coefficient_correction

**This is a very minor bug when n is large** as bessel_coefficient_correction tends to 1. But when number of elements per channel is small it becomes noticeable. Below I gave an example code for when elements per channel is 2.

**Describe the expected behavior**

I would expect `fused=True` and `fused=False` to be consistent.

- Do you want to contribute a PR? (yes/no): no

**Solution**

I think the issue lies [here](https://github.com/tensorflow/tensorflow/blob/b89a2f2a1b761ac68aa808f2bcd847314b15c6c1/tensorflow/core/kernels/fused_batch_norm_op.cc#L213). In which you can see the bessel_coefficient_correction as `rest_size_adjust`. Either that should be removed in fused batch norm or it should be added for non-fused batch norm.

**Standalone code to reproduce the issue**

```
import tensorflow as tf
from math import sqrt


with tf.device("/cpu:0"):
# with tf.device("/gpu:0"):
    var = 5.0
    std = sqrt(var)
    n=2
    inp = tf.random.normal((1,n,1,1))
    true_mean = tf.reduce_mean(inp, axis=(0, 1, 2)).numpy()
    print("True mean: " + str(true_mean))
    true_var = tf.math.reduce_variance(inp, axis=(0, 1, 2)).numpy()
    print("True variance: " + str(true_var))

    for momentum in [0.0, 0.25, 0.75, 0.9]:
        for fused in [True, False]:
            layer = tf.keras.layers.BatchNormalization(
                scale=True, center=True, trainable=True, momentum=momentum, fused=fused
            )
            initial_mean = 0.0
            initial_var = 1.0
            out = layer(inp, training=True)
            print("\n===========================")
            print("Momentum: " + str(momentum) + " Fused: " + str(fused))
            print("Moving mean: " + str(layer.moving_mean.numpy()))
            print(
                "Moving mean always correctly computed: "
                + str(initial_mean * momentum + true_mean * (1 - momentum))
            )
            print(f"Moving var computed with fused={fused}: " + str(layer.moving_variance.numpy()))
            print(
                "Moving var with Bessel correction: "
                + str(initial_var * momentum + n/(n-1) * true_var * (1 - momentum))
            )
            print(
                "Moving var without Bessel correction: "
                + str(initial_var * momentum + true_var * (1 - momentum))
            )
            print("===========================\n")

```

See this gist: https://gist.github.com/canbakiskan/9a1fe01a218277539f9602081719a1c3

**Comment by tilakrayal** [comment]
@sachinprasadhs ,
I was able to reproduce the issue in tf v2.8, v2.7 and nightly.Please find the gist [here](https://colab.research.google.com/gist/tilakrayal/185c8c0b3a40c26d8e36542210fc1be4/55545.ipynb).

**Comment by sachinprasadhs** [comment]
You can find some details when fused is set to True. Let us know if this helps you.
https://github.com/keras-team/keras/blob/d8fcb9d4d4dad45080ecfdd575483653028f8eda/keras/layers/normalization/batch_normalization.py#L197-L205

**Comment by canbakiskan** [comment]
Not really, the problem isn't when `fused` is used but rather what happens when it is used is wrong. Let me also point out that `fused=True` is the default behavior.

**Comment by canbakiskan** [comment]
Ok I found the culprit. It's not that the batch's variance is multiplied by 2 but rather to Bessel's coefficient correction which is n/(n-1) where n is the number of elements in a channel of the tensor [here](https://github.com/tensorflow/tensorflow/blob/b89a2f2a1b761ac68aa808f2bcd847314b15c6c1/tensorflow/core/kernels/fused_batch_norm_op.cc#L213).

It's still inconsistent behavior with respect to `fused=False` case though. I'm changing the title and first post accordingly.

**Comment by sachinprasadhs** [comment]
Development of keras moved to separate repository https://github.com/keras-team/keras/issues

Please post this issue on keras-team/keras repo.
To know more see;
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!
